sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -252,7 +252,7 @@ trait Row extends Serializable {
    *
    * @throws ClassCastException when data type does not match.
    */
-  def getDate(i: Int): java.sql.Date = apply(i).asInstanceOf[java.sql.Date]
+  def getDate(i: Int): java.sql.Date = DateUtils.toJavaDate(getInt(i))
```

Comment:
Now I add the conversion in DataTypeConversion, which is only valid to java class. For scala, We need to write
Row(DateUtils.fromJavaDate(...)). Is this OK with you?